lib/zlib.js

```diff
@@ -251,9 +251,20 @@ function Zlib(opts, mode) {
       opts.finishFlush, 'options.finishFlush',
       Z_NO_FLUSH, Z_BLOCK, Z_FINISH);
 
-    windowBits = checkRangesOrGetDefault(
-      opts.windowBits, 'options.windowBits',
-      Z_MIN_WINDOWBITS, Z_MAX_WINDOWBITS, Z_DEFAULT_WINDOWBITS);
+      // windowBits is special. On the compression side, 0 is an invalid value.
+      // But on the decompression side, a value of 0 for windowBits tells zlib
+      // to extract the value from the headers of the compressed stream.
+    if (opts.windowBits === 0 &&
+        (mode === INFLATE ||
+         mode === GUNZIP ||
+         mode === INFLATERAW ||
```

Comment:
`INFLATERAW` means that there is no header, so I think we'd want to drop that case?

Reply:
Sure. I picked the list from https://github.com/nodejs/node/blob/22f4a35db344472db1e83f9e3156907b58f5f527/src/node_zlib.cc#L541-L544, since they all end up calling `inflateInit2()`. Will remove it from the list.